ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java

```diff
@@ -30,477 +33,299 @@
 import java.util.List;
 import java.util.Properties;
 
+import static com.spectralogic.ds3cli.ArgumentFactory.*;
+
 public class Arguments {
 
     private final static String PROPERTY_FILE = "ds3_cli.properties";
 
+    // variables required by Arguments class
+    private final String[] args;
     private final Options options;
+    private CommandLine cmd;
+    private final List<String> missingArgs = new ArrayList<>();
 
-    private String bucket;
-    private String directory;
-    private String endpoint;
-    private String accessKey;
-    private String secretKey;
-    private String prefix;
-    private String id = null;
-    private String command;
+    // locally stored properties processed in initial parsing
+    // useful to provide defaults and handle parsing exceptions
+    // most simple values are queried as needed
```

Comment:
?? Thanks -- I agree with everything else and will fix but don't quite get this one.